async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -392,6 +397,7 @@ pub struct Stream<'a> {
     subject: String,
     batch_config: BatchConfig,
     request: Option<BoxFuture<'a, Result<(), Error>>>,
+    since_last_request: Arc<Mutex<Instant>>,
```

Comment:
Don't have to point this out but, obviously preferable to not have a mutex here.
But for a fix, we can allow it with the intent to iterate soon.

Reply:
Yeah, we could go with pending as atomic shared between task and iterator fn, with Instant hold on task only and having task only sending requests, but that also needs returning error from it via channel to iterator. I would prefer revisit it soon and now focus on more important work.